kci_build

```diff
@@ -304,8 +306,9 @@ or
             return False
 
         step = kernelci.build.RevisionData(args.kdir, args.output, reset=True)
-        print("Initialising build meta-data in {}".format(args.output))
-        res = step.run(tree_name, tree_url, branch)
+        print("Initialising build meta-data in {}".format(step.output_path))
+        res = step.run(tree_name, tree_url, branch,
```

Comment:
Well, I'm not sure how to do that in such a way that it would help.  Each build step takes a different set of arguments, so we could have a dictionary argument and document what goes inside but that would seem like more code complexity for handling something Python already does (i.e. keyword arguments). Simply having `(self, *args, **kwargs)` in the base class would not tell how derived class are used, and they would still have different method signatures.

Reply:
At the moment the base class says clearly that the method takes no arguments. I'd say it'd be better to indicate that it actually can take arbitrary number of arguments.

Reply:
Fair enough, I'll update the docstring too.

Reply:
I've replaced the `Step.run()` signature and all its derived classes with this:
```python
def run(self, jopt=None, verbose=False, opts=None):
```
Now the `opts` argument is a dictionary with arbitrary options specific to each step implementation.